### PR TITLE
enable kafka broker SSL config and spring kafka annotation

### PR DESCRIPTION
**Pull request by chengjie8**
## What was the problem?
- At present, there's no system in place for the VRO platform to receive BIE Kafka topic messages across different environments.

Associated tickets or Slack threads:
- #1682 

## How does this fix it?[^1]
- The code allows the svc-bie-kafka microservice to establish a connection with the BIE Kafka broker and consume messages from five different topics related to contention classification.

## How to test this PR
- Presently, the Kafka consumer code, which is configured with SSL, cannot be tested locally due to the need for VA network access. Nevertheless, a subsequent PR could introduce a potential mock Kafka producer method. This, coupled with the mock Kafka broker, might provide a solution for testing the existing code.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: svc-bie-kafka/src/main/resources/application-sandbox.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+spring:
+  # Specify where Spring connects to Kafka
```
Comment (tejans24) [inline svc-bie-kafka/src/main/resources/application-sandbox.yaml:2]: can be removed

File: svc-bie-kafka/src/main/java/gov/va/vro/services/bie/service/kafka/KafkaConsumer.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class KafkaConsumer {
+  private final AmqpMessageSender amqpMessageSender;
+  private final BieProperties bieProperties;
+
+  @KafkaListener(
+      topics = {
+        "#{'${kafka.topic.prefix}'}_CONTENTION_BIE_CONTENTION_ASSOCIATED_TO_CLAIM_V02",
```
Comment (tejans24) [inline svc-bie-kafka/src/main/java/gov/va/vro/services/bie/service/kafka/KafkaConsumer.java:22]: nice this seems much cleaner, should these be in properties (`kafka-topic-to-amqp-exchange-map`) or within the code here?

**Review by tejans24 — COMMENTED** [review]
LGTM with a couple of comments, tests seem to be failing as well

File: settings.gradle [hunk 3]
```diff
@@ -16,13 +16,13 @@ pluginManagement {
...
 }
 
-
 dependencyResolutionManagement {
   repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
   repositories {
     // https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:case-for-maven-local
     // Not recommended: mavenLocal()
     maven { url "https://repo1.maven.org/maven2/" }
+    maven { url "https://packages.confluent.io/maven/" }
```
Comment (tejans24) [inline settings.gradle:25]: I'm guessing this hasn't been an issue with secrel?

**Review by tejans24 — APPROVED** [review]

File: svc-bie-kafka/src/main/java/gov/va/vro/services/bie/service/kafka/KafkaConsumer.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      String messageValue = new String(record.value(), StandardCharsets.UTF_8);
+      String topicName = record.topic();
+
+      log.info("Topic name: {}", topicName);
+      log.info("Consumed message key: {}", messageKey);
+      log.info("Consumed message value (before) decode: {}", messageValue);
+
+      amqpMessageSender.send(
+          bieProperties.getKafkaTopicToAmqpExchangeMap().get(topicName), topicName, messageValue);
+    } catch (Exception e) {
```
Comment (engineer-plus-plus) [inline svc-bie-kafka/src/main/java/gov/va/vro/services/bie/service/kafka/KafkaConsumer.java:40]: Logging of  exceptions should be done by the service's exception-handling module not the business logic.  Single responsiblity and cohesion imply so.  I would just let exceptions go in this case.

File: svc-bie-kafka/src/test/java/gov/va/vro/services/bie/config/KafkaConsumerCreatorTest.java [hunk 5]
```diff
@@ -1,53 +1,24 @@
...
-
 @ExtendWith(MockitoExtension.class)
 class KafkaConsumerCreatorTest {
 
   @Mock private ConsumerFactory<?, ?> consumerFactory;
   @Mock private AmqpMessageSender amqpMessageSender;
 
   private BieProperties bieProperties;
-  private KafkaConsumerCreator kafkaConsumerCreator;
+  private KafkaConsumer kafkaConsumer;
```
Comment (engineer-plus-plus) [inline svc-bie-kafka/src/test/java/gov/va/vro/services/bie/config/KafkaConsumerCreatorTest.java:18]: souldn't you change the file/class name to too?

**Review by engineer-plus-plus — CHANGES_REQUESTED** [review]

File: svc-bie-kafka/src/test/java/gov/va/vro/services/bie/config/KafkaConsumerCreatorTest.java (hunk 5 above)
Comment (chengjie8) [reply svc-bie-kafka/src/test/java/gov/va/vro/services/bie/config/KafkaConsumerCreatorTest.java]: Addressed

File: svc-bie-kafka/src/main/java/gov/va/vro/services/bie/service/kafka/KafkaConsumer.java (hunk 4 above)
Comment (chengjie8) [reply svc-bie-kafka/src/main/java/gov/va/vro/services/bie/service/kafka/KafkaConsumer.java]: The try catch was for schema registry decoding kafka message using schema file, the code is moving to another PR, so I clean up the try catch here.

**Review by greene858 — APPROVED** [review]
I would like to see coverage of KafkaConsume, please make sure we have a ticket to add

**Pull request merged**